File: src/tts.py
import pyttsx3

import constants
import os


__engine = pyttsx3.init()


def say_message(message):
    """
    Says <message> using TTS engine.

    This is used for fast debugging when testing
    discord bot's ability to read and speak messages.
    """
    __engine.say(message)


def __clear_audio():
    """
    Clears audio.
    Used before updating the audio.
    """
    try:
        os.remove(constants.AUDIO_FILE_PATH)

    except FileNotFoundError:
        print(f"File '{constants.AUDIO_FILE_PATH}' doesn't exist.")


def create_tts_audio_file(message):
    """
    Using the text in <message>, this creates a TTS (text-to-speech) mp3 file and stores 
    the file in "audio_file\sounds.mp3".
    """
    __clear_audio()

    __engine.save_to_file(message, constants.AUDIO_FILE_PATH)
    __engine.runAndWait()

File: src/tts.rs
use delete::{delete_file};

use gtts::save_to_file;


fn clear_audio(file_path: &str) {
    // Deletes Sound.mp3.
    // Ran before creating an audio file.

    delete_file(file_path).unwrap();
}

pub fn create_audio(message: &str) {
    let file_path: &str = r"audio_files\Sound.mp3";
    clear_audio(&file_path);

    // Creates an audio_files\Sound.mp3.
    // This mp3 file contains the TTS of the message.
    save_to_file(message, &file_path);
}